### PR TITLE
[ie/art19] Add extractors

### PR DESCRIPTION
**Pull request by seproDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Adds an art19 extractor, which can then serve as a basis for #8875 and #8838. Both are currently not dealt with due to their non-standard way to embed the podcast.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/art19.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import float_or_none, int_or_none, parse_iso8601, url_or_none
+from ..utils.traversal import traverse_obj
+
+
+class Art19IE(InfoExtractor):
+    _UUID_REGEX = r'[\da-f]{8}-?[\da-f]{4}-?[\da-f]{4}-?[\da-f]{4}-?[\da-f]{12}'
```
Comment (bashonly) [inline yt_dlp/extractor/art19.py:9]: I think a UUID without dashes is not valid. Unless you have examples of this site using invalid UUIDs, I would simplify the pattern by enforcing dashes
```suggestion
    _UUID_REGEX = r'[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}'
```
&nbsp;
could do this to make it shorter, although arguably harder to read:
```suggestion
    _UUID_REGEX = r'[\da-f]{8}(?:-[\da-f]{4}){3}-[\da-f]{12}'
```
really this should be a util by now imo

File: yt_dlp/extractor/art19.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    _VALID_URL = [
+        rf'https?://(?:www\.)?art19\.com/shows/[^/#?]+/episodes/(?P<id>{_UUID_REGEX})',
+        rf'https?://rss\.art19\.com/episodes/(?P<id>{_UUID_REGEX})\.mp3',
+    ]
+    _EMBED_REGEX = [rf'(?x)<iframe[^>]+\bsrc=[\'"](?P<url>{_VALID_URL[0]})']
```
Comment (bashonly) [inline yt_dlp/extractor/art19.py:14]: not needed
```suggestion
    _EMBED_REGEX = [rf'<iframe[^>]+\bsrc=[\'"](?P<url>{_VALID_URL[0]})']
```

File: yt_dlp/extractor/art19.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+        }
+
+
+class Art19ShowIE(InfoExtractor):
+    _VALID_URL_BASE = r'https?://(?:www\.)?art19\.com/shows/(?P<id>[\w-]+)(?:/embed)?/?'
+    _VALID_URL = [
+        rf'{_VALID_URL_BASE}(?:$|[#?])',
+        r'https?://rss\.art19\.com/(?P<id>[\w-]+)/?(?:$|[#?])',
+    ]
+    _EMBED_REGEX = [rf'(?x)<iframe[^>]+\bsrc=[\'"](?P<url>{_VALID_URL_BASE}[^\'"])']
```
Comment (bashonly) [inline yt_dlp/extractor/art19.py:181]:
```suggestion
    _EMBED_REGEX = [rf'<iframe[^>]+\bsrc=[\'"](?P<url>{_VALID_URL_BASE}[^\'"])']
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/art19.py (hunk 1 above)
Comment (seproDev) [reply yt_dlp/extractor/art19.py:9]: The site accepts both 
`https://art19.com/shows/scamfluencers/episodes/8319b776-4153-4d22-8630-631f204a03dd` (with all dashes)
and
`https://art19.com/shows/scamfluencers/episodes/8319b77641534d228630631f204a03dd` (no dashes)
but not partial dashes or dashes in different places

File: yt_dlp/extractor/art19.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                'vcodec': 'none',
+                'acodec': fmt_id,
+                'quality': -2 if fmt_id == 'ogg' else -1,
+            })
+        formats.append({
+            'format_id': 'direct',
+            'url': f'https://rss.art19.com/episodes/{episode_id}.mp3',
+            'vcodec': 'none',
+            'acodec': 'mp3',
+        })
```
Comment (bashonly) [inline yt_dlp/extractor/art19.py:143]:
```suggestion
        formats = [{
            'format_id': 'direct',
            'url': f'https://rss.art19.com/episodes/{episode_id}.mp3',
            'vcodec': 'none',
            'acodec': 'mp3',
        }]
        for fmt_id, fmt_data in traverse_obj(rss_metadata, ('content', 'media', {dict.items}, ...)):
            if fmt_id == 'waveform_bin':
                continue
            fmt_url = traverse_obj(fmt_data, ('url', {url_or_none}))
            if not fmt_url:
                continue
            formats.append({
                'format_id': fmt_id,
                'url': fmt_url,
                'vcodec': 'none',
                'acodec': fmt_id,
                'quality': -2 if fmt_id == 'ogg' else -1,
            })
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**